community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/ast/convert/plannerQuery/PlannerQueryBuilder.scala

```diff
@@ -44,8 +44,14 @@ case class PlannerQueryBuilder(private val q: PlannerQuery, semanticTable: Seman
       case (acc, next) if next.horizon.exposedSymbols(next.queryGraph).nonEmpty => next
       case (acc, _) => acc
     }
-    lastPQWithNonEmptyHorizon.horizon.exposedSymbols(lastPQWithNonEmptyHorizon.queryGraph) ++
-      lastPQWithNonEmptyHorizon.queryGraph.allCoveredIds
+
+    val horizon = lastPQWithNonEmptyHorizon.horizon
+    val queryGraph = lastPQWithNonEmptyHorizon.queryGraph
+    val exposed = horizon.exposedSymbols(queryGraph)
+    horizon match {
```

Comment:
Maybe. But this was backwards since the beginning since this code here was adding the `queryGraph.allCoveredIds` to the `exposedSymbols` from the horizon.